### PR TITLE
Pro 2282 pro 2236 import button and modal

### PR DESCRIPTION
**Pull request by ValJed**
[PRO-2182](https://linear.app/apostrophecms/issue/PRO-2182/as-an-editor-i-can-click-a-button-to-open-a-modal-for-importing-pieces)
[PRO-2236](https://linear.app/apostrophecms/issue/PRO-2236/as-an-editor-i-am-presented-with-a-modal-screen-to-upload-my-file-for)

## Purpose
Adds a new component `AposFile.vue` to abstract file dropzone UI. Uses this new component in `AposInputAttachment`. And uses it in the pieces import modal.

## How to test
Check that normal attachment UI is still working as expected.
Test the new import modal by trying to import csv files. (confirm button does nothing for now).
Import can be test with y linking this module's PR: https://github.com/apostrophecms/piece-type-importer/pull/1
And the `testbed` project has an `import` branch with this new module configured.

### CODE REVIEW

File: modules/@apostrophecms/i18n/i18n/en.json
Comment (ValJed) [inline modules/@apostrophecms/i18n/i18n/en.json:153]: Not sure about this one since it wasn't complete in designs..

File: modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue
Comment (ValJed) [inline modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue:269]: Ideally this method would be a `andleUtilityOperation` generic one, but since we are using specific component, managing file... it's not that easy and I didn't have enough time to manage this cleanly. So maybe we could discuss it and create ticket for it.

Comment (boutell) [reply modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue:269]: per discussion this needs to be fixed before the PR lands, we arrived at a plan.

**Review by boutell — CHANGES_REQUESTED** [review]
Per discussion there is a plan to fix it so there is no import-specific logic in core (factoring out AposFile is OK).

File: modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue
Comment (ValJed) [reply modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue]: Has been changes in `handleUtilityOperation`, now it manages new modals or confirmation modal as before, support for form should maybe be added but not sure what possibilities should be available.

**Review by abea — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/task/index.js
Comment (ValJed) [reply modules/@apostrophecms/task/index.js:222]: You're right, that wasn't my code I just fixed a linter error with the new eslint rule. modified.

File: modules/@apostrophecms/piece-type/index.js
Comment (ValJed) [reply modules/@apostrophecms/piece-type/index.js:99]: Yes removed

File: modules/@apostrophecms/piece-type/index.js
Comment (ValJed) [reply modules/@apostrophecms/piece-type/index.js:104]: Ok, I removed localization for now, and since I don't know how to do it from another module I make it in suspend.

File: .eslintrc
Comment (ValJed) [reply .eslintrc:15]: Ok I modified the rule, but I think when using lodash, it would be great to import the utility we need only instead of the whole lib.

File: modules/@apostrophecms/ui/ui/apos/components/AposFile.vue
Comment (ValJed) [reply modules/@apostrophecms/ui/ui/apos/components/AposFile.vue:26]: Yes I was too.. but I think you're tight, since there is no place where we use a multi files dropzone for now it makes no sense to allow it. + It wouldn't be that complicated to modify if needed..

File: modules/@apostrophecms/ui/ui/apos/components/AposFile.vue
Comment (ValJed) [reply modules/@apostrophecms/ui/ui/apos/components/AposFile.vue:118]: Ok, this part was unclear since it was passing `field.accept` to the input directly but in another hand, an array of allowedExtensions was allowing all extensions ad was checked from another method. To make it clearer and less scattered i'll use the string only and use this one in the `checkFileGroup` method. `groupInfo.extensions` was empty everytime I checked.

Comment (ValJed) [reply modules/@apostrophecms/ui/ui/apos/components/AposFile.vue:118]: Also accept is not enough since it doesn't block bad type files from drag and drop.

File: modules/@apostrophecms/ui/ui/apos/components/AposFile.vue
Comment (ValJed) [reply modules/@apostrophecms/ui/ui/apos/components/AposFile.vue:125]: Because AposSlatList and AposSlat need an `_id`, that was the most confusing part for me because these components seem to be able to manage multiple items that must be attachments but in our case we manage only one, so the update method with items for example is confusing. This fakeId allowed me to display the selected file and allow to remove it without modifying any code from `AposSlatList` or `AposSlat`, I didn't want to take any side effect risk. (and they seem complex enough to avoid that).

File: modules/@apostrophecms/ui/ui/apos/components/AposFile.vue
Comment (ValJed) [reply modules/@apostrophecms/ui/ui/apos/components/AposFile.vue:26]: Modified to allow one selected file only.

File: modules/@apostrophecms/i18n/i18n/en.json
Comment (ValJed) [reply modules/@apostrophecms/i18n/i18n/en.json]: I remove them.

File: modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue
Comment (boutell) [reply modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue:277]: Hmm, there's nothing that needs to happen after it, but awaiting it is best yes in case someone invoking this method (which claims to be async, after all) cares about timing.

File: modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue
Comment (boutell) [reply modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue:285]: That does seem likely.

File: modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue
Comment (boutell) [inline modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue:295]: Standard convention elsewhere in apostrophe is to insert the "/" between the action and route name here.

In particular the developer should NOT write "route: '/archive'" to indicate a route called archive in the current module. They should just write "route: 'archive'."

Otherwise it is very ambiguous what is meant by the "/".

If we do this right it will be possible at any point to support arbitrary URLs as well because we can immediately tell by the leading slash that they do *not* want the action prepended to them.

File: modules/@apostrophecms/ui/ui/apos/components/AposFile.vue
Comment (boutell) [reply modules/@apostrophecms/ui/ui/apos/components/AposFile.vue:26]: Agree this is OK for now.

**Review by boutell — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue
Comment (abea) [reply modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue:285]: We can hold off on this for later.

File: modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue
Comment (abea) [reply modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue:277]: This should still be updated for clarity.

File: modules/@apostrophecms/ui/ui/apos/components/AposFile.vue
Comment (abea) [reply modules/@apostrophecms/ui/ui/apos/components/AposFile.vue]: 😮 I see. And the attachment input had a real _id because it uploaded the file to the attachments collection.

File: modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue
Comment (abea) [reply modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue:295]: This is related to how I originally implemented batch operations. @ValJed I changed that to use the batch operation name and removed the `route` option completely.

**Review by abea — CHANGES_REQUESTED** [review]
One `await` for me in addition to Tom's comments.

File: modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue
Comment (ValJed) [reply modules/@apostrophecms/piece-type/ui/apos/components/AposDocsManager.vue:295]: Ok no problem I change that.

**Review by boutell — APPROVED** [review]

**Pull request merged**
